### PR TITLE
Meets Bug #3531: Type "None" cannot be configured via Admin settings

### PR DESCRIPTION
**Pull request by netfighter**
Adds standard types to the configuration list of types
Forbids destroy and name change for the standard types
Adds a cucumber scenario that tests the new functionality
Adds a spec that tests all the methods from the TypesController

### CODE REVIEW

File: features/step_definitions/type_steps.rb [hunk 1]
```diff
@@ -45,4 +45,4 @@
 
 Then /^I should not see the "([^"]*)" type$/ do |name|
   page.all(:css, '.timelines-pet-name', :text => name).should be_empty
-end
+end
```
Comment (marutosi) [inline features/step_definitions/type_steps.rb:48]: unrelated change

File: spec/controllers/types_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        expect(response.body).to have_content("Name can't be blank")
+      end
+    end
+
+    describe "WITH workflow copy" do
+      let!(:existing_type) { FactoryGirl.create(:type, :name => 'Existing type') }
+      let!(:workflow) { FactoryGirl.create(:workflow,
+                                          old_status: status_0,
+                                          new_status: status_1,
+                                          type_id: existing_type.id) }
```
Comment (hschink) [inline spec/controllers/types_controller_spec.rb]: Indentation

File: spec/controllers/types_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+      } }
+
+      before do
+        post :create, params
+      end
+
+      it { expect(response).to be_redirect }
+      it { expect(response).to redirect_to(types_path) }
+      it 'should have the copied workflows' do
+        Type.find_by_name('New type').workflows.count.should eq(existing_type.workflows.count)
```
Comment (hschink) [inline spec/controllers/types_controller_spec.rb]: `expect` syntax

File: spec/controllers/types_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+    before do
+      get 'edit', :id => type.id
+    end
+
+    it { expect(response).to be_success }
+    it { expect(response).to render_template 'edit' }
+    it { response.body.should have_selector "input[@name='type[name]'][@value='My type']" }
+    it { response.body.should have_selector "input[@name='type[project_ids][]'][@value='#{project.id}'][@checked='checked']" }
+    it { response.body.should have_selector "input[@name='type[is_milestone]'][@value='1'][@checked='checked']" }
```
Comment (hschink) [inline spec/controllers/types_controller_spec.rb]: `expect` syntax: `expect(response.body).to have_selector "input[@name='type[name]'][@value='My type']"`

File: spec/controllers/types_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+      let(:params) { { 'id' => type.id, 'type' => { :name => 'My type renamed' } } }
+
+      before do
+        put :update, params
+      end
+
+      it { expect(response).to be_redirect }
+      it { expect(response).to redirect_to(types_path) }
+      it 'should be renamed' do
+        Type.find_by_name('My type renamed').id.should eq(type.id)
```
Comment (hschink) [inline spec/controllers/types_controller_spec.rb]: see above

File: spec/controllers/types_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+      let(:params) { { 'id' => type.id, 'type' => { :project_ids => [''] } } }
+
+      before do
+        put :update, params
+      end
+
+      it { expect(response).to be_redirect }
+      it { expect(response).to redirect_to(types_path) }
+      it 'should have no projects assigned' do
+        Type.find_by_name('My type').projects.count.should eq(0)
```
Comment (hschink) [inline spec/controllers/types_controller_spec.rb]: see above
